csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -212,6 +212,96 @@ private void CountSelfAndChildrenNodes(IArrowType type, ref int count)
             count++;
         }
 
+        private protected void WriteRecordBatchInternal(RecordBatch recordBatch)
+        {
+            // TODO: Truncate buffers with extraneous padding / unused capacity
+
+            if (!HasWrittenSchema)
+            {
+                WriteSchema(Schema);
+                HasWrittenSchema = true;
+            }
+
+            Builder.Clear();
```

Comment:
Do you think the next ~40 lines of code could be refactored out so they weren't duplicate?

Reply:
Refactored this part out.  Let me know if this was what you were expecting.

Reply:
Yep - this is what I had in mind.